frontend/app/src/features/room/components/code/CodeEditor.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react'
+import CodeMirror, { basicSetup } from '@uiw/react-codemirror'
+import { Socket } from "socket.io-client"
+import { LanguageSupport, indentUnit } from '@codemirror/language'
+import { Box, Text } from '@chakra-ui/react'
+import { getDocument, peerExtension } from '@/lib/collab'
+import { Spinner } from '@chakra-ui/react'
+
+interface CodeEditorProps {
+    socket: Socket;
+    className?: string;
+    roomId: string;
+    language: LanguageSupport;
+}
+
+export const CodeEditor = ({ socket, roomId, language }: CodeEditorProps) => {
+
+    const [doc, setDoc] = useState<string | null>(null)
+    const [version, setVersion] = useState<number | null>(null)
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const { version, doc } = await getDocument(socket, roomId)
+                setVersion(version)
+                setDoc(doc.toString())
+            } catch (error) {
+                console.error('Error fetching document', error)
+            }
+        }
+
+        fetchData()
+
+        return () => {
+            socket.off('connect')
+            socket.off('disconnect')
+            socket.off('pullUpdateResponse')
+            socket.off('pushUpdateResponse')
+            socket.off('getDocumentResponse')
+        }
+
+    }, [socket, roomId])
+
+
+
+    if (version != null && doc != null) {
```

Comment:
strict equality check
```suggestion
    if (version !== null && doc !== null) {
```

Reply:
resolved in [b6ccf98](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/pull/18/commits/b6ccf98a9fb7fb920d30958b64457b7d5f868781)